mitty/test/simulation/genome/test_sampledgenome.py

```diff
@@ -0,0 +1,31 @@
+"""Tests for the generation of a random individual from given vcf"""
+import os
+
+import pysam
+import mitty.test
+import mitty.simulation.genome.sampledgenome as sg
+
+
+def read_data(outname):
+  vcf_pointer = pysam.VariantFile(filename=outname)
+  sample = vcf_pointer.header.samples[0]
+  gts = []
+  for rec in vcf_pointer.fetch():
+    gts.append("/".join([str(ind) for ind in rec.samples["HG"].allele_indices]))
+  return sample, gts
+
+
+def test_assign_random_gt():
+  outname = os.path.join(mitty.test.example_data_dir, 'random_gt.vcf')
```

Comment:
Convert `sg.assign_random_gt` to use a file pointer (please see elsewhere) and then use `tempfile` to create a temporary file, instead of writing to the test directory.